## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :x: [more...](#bab2b0a5-4abf-4dc2-a93f-3ad3e0f4bbd2) |
<a name="bab2b0a5-4abf-4dc2-a93f-3ad3e0f4bbd2"></a>
### 16176-6e07836/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.7:30516
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 2f7ffb6f0cd4:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... FAILED in 2 seconds
Dec 06, 2022 11:45:07 AM com.amazonaws.util.Base64 warn
WARNING: JAXB is unavailable. Will fallback to SDK implementation which may be less performant.If you are using Java 9+, you will need to include javax.xml.bind:jaxb-api as a dependency.
Exception in thread "main" com.amazonaws.services.s3.model.AmazonS3Exception: We encountered an internal error, please try again.: cause(listPathRaw: 0 drives provided) (Service: Amazon S3; Status Code: 500; Error Code: InternalError; Request ID: 172E31A7BB0EFD8B; S3 Extended Request ID: e8483dec-32c7-4ced-8603-c81f3c3ee53c), S3 Extended Request ID: e8483dec-32c7-4ced-8603-c81f3c3ee53c
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1742)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1371)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1347)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1127)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:784)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:752)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5052)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4998)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4992)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:895)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:863)
	at io.minio.awssdk.tests.FunctionalTests.teardown(Unknown Source)
	at io.minio.awssdk.tests.FunctionalTests.main(Unknown Source)
(2/14) Running aws-sdk-php tests ... done in 43 seconds
(3/14) Running aws-sdk-ruby tests ... done in 7 seconds
(4/14) Running awscli tests ... FAILED in 20 seconds
{
  "name": "awscli",
  "duration": 10215,
  "function": "aws --endpoint-url http://15.15.15.7:30516 s3api list-objects --bucket awscli-mint-test-bucket-10765 --prefix datafile-1-kB\n",
  "status": "FAIL",
  "error": ""
}
(4/14) Running healthcheck tests ... done in 0 seconds
(5/14) Running mc tests ... done in 1 minutes and 11 seconds
(6/14) Running minio-go tests ... FAILED in 7 seconds
{
  "args": {
    "bucketName": "minio-go-test-9kdy5nzei5q3mo63",
    "isRecursive": true,
    "objectName": "minio-go-test-9kdy5nzei5q3mo63unique"
  },
  "duration": 4693,
  "function": "ListObjects(bucketName, objectName, isRecursive, doneCh)",
  "message": "Could not find existing object minio-go-test-9kdy5nzei5q3mo63unique",
  "name": "minio-go: testFunctionalV2",
  "status": "FAIL"
}
(6/14) Running minio-java tests ... done in 34 seconds
(7/14) Running minio-js tests ... FAILED in 46 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook for \"removeObjects(bucketName, objectsList)",
  "args": "bucketName:minio-js-test-ver-presign-300235de-9e59-4bad-bf52-978cd72d55f6\"",
  "duration": 7,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(7/14) Running minio-py tests ... FAILED in 42 seconds
{
  "name": "minio-py:test_list_objects_v1",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-160c366a-0668-4169-8a52-36c1a37ab974",
    "object_name": "101d2430-5a2b-4b41-8534-280de0ed7f7c",
    "recursive": true
  },
  "message": "HTTPConnectionPool(host='15.15.15.7', port=30516): Max retries exceeded with url: /minio-py-test-160c366a-0668-4169-8a52-36c1a37ab974?delimiter=&encoding-type=url&max-keys=1000&prefix= (Caused by ResponseError('too many 500 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1107, in test_list_objects_v1\n    _test_list_objects(log_entry, use_api_v1=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 1086, in _test_list_objects\n    for obj in objects:\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 2736, in _list_objects\n    response = self._execute(\"GET\", bucket_name, query_params=query)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.7', port=30516): Max retries exceeded with url: /minio-py-test-160c366a-0668-4169-8a52-36c1a37ab974?delimiter=&encoding-type=url&max-keys=1000&prefix= (Caused by ResponseError('too many 500 error responses'))\n",
  "duration": 6224
}
(7/14) Running s3cmd tests ... done in 25 seconds
(8/14) Running s3select tests ... done in 4 seconds
(9/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 9 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/xl-storage.go
+++ b/cmd/xl-storage.go
@@ -2329,35 +2329,42 @@ func (s *xlStorage) RenameData(ctx context.Context, srcVolume, srcPath string, f
 		}
 	}
 
-	var oldDstDataPath string
+	var oldDstDataPath, reqVID string
+
 	if fi.VersionID == "" {
-		// return the latest "null" versionId info
-		ofi, err := xlMeta.ToFileInfo(dstVolume, dstPath, nullVersionID)
-		if err == nil && !ofi.Deleted {
-			if xlMeta.SharedDataDirCountStr(nullVersionID, ofi.DataDir) == 0 {
-				// Purge the destination path as we are not preserving anything
-				// versioned object was not requested.
-				oldDstDataPath = pathJoin(dstVolumeDir, dstPath, ofi.DataDir)
-				// if old destination path is same as new destination path
-				// there is nothing to purge, this is true in case of healing
-				// avoid setting oldDstDataPath at that point.
-				if oldDstDataPath == dstDataPath {
-					oldDstDataPath = ""
-				}
-				xlMeta.data.remove(nullVersionID, ofi.DataDir)
+		reqVID = nullVersionID
+	} else {
+		reqVID = fi.VersionID
+	}
+
+	// Replace the data of null version or any other existing version-id
+	ofi, err := xlMeta.ToFileInfo(dstVolume, dstPath, reqVID)
+	if err == nil && !ofi.Deleted {
+		if xlMeta.SharedDataDirCountStr(reqVID, ofi.DataDir) == 0 {
+			// Purge the destination path as we are not preserving anything
+			// versioned object was not requested.
+			oldDstDataPath = pathJoin(dstVolumeDir, dstPath, ofi.DataDir)
+			// if old destination path is same as new destination path
+			// there is nothing to purge, this is true in case of healing
+			// avoid setting oldDstDataPath at that point.
+			if oldDstDataPath == dstDataPath {
+				oldDstDataPath = ""
+			} else {
+				xlMeta.data.remove(reqVID, ofi.DataDir)
 			}
 		}
-		// Empty fi.VersionID indicates that versioning is either
-		// suspended or disabled on this bucket. RenameData will replace
-		// the 'null' version. We add a free-version to track its tiered
-		// content for asynchronous deletion.
-		if !fi.IsRestoreObjReq() {
-			// Note: Restore object request reuses PutObject/Multipart
-			// upload to copy back its data from the remote tier. This
-			// doesn't replace the existing version, so we don't need to add
-			// a free-version.
-			xlMeta.AddFreeVersion(fi)
-		}
+	}
+
+	// Empty fi.VersionID indicates that versioning is either
+	// suspended or disabled on this bucket. RenameData will replace
+	// the 'null' version. We add a free-version to track its tiered
+	// content for asynchronous deletion.
+	if fi.VersionID == "" && !fi.IsRestoreObjReq() {
+		// Note: Restore object request reuses PutObject/Multipart
+		// upload to copy back its data from the remote tier. This
+		// doesn't replace the existing version, so we don't need to add
+		// a free-version.
+		xlMeta.AddFreeVersion(fi)
 	}
 
 	// indicates if RenameData() is called by healing.

